src/iconv.jl

```diff
@@ -1,9 +1,61 @@
 # This file is a part of Julia. License is MIT: http://julialang.org/license
 
 module iconv
-import Base: close, eof, flush, read, readall, write
+import Base: close, eof, flush, read, readall, write, show
 import Base.Libc: errno, strerror, E2BIG, EINVAL, EILSEQ
 export StringEncoder, StringDecoder, encode, decode
+export StringEncodingError, OutputBufferError, IConvError
+export InvalidEncodingError, InvalidSequenceError, IncompleteSequenceError
+
+
+abstract StringEncodingError
+
+# specified encodings are not supported
+type InvalidEncodingError <: StringEncodingError
+    msg::UTF8String
```

Comment:
I tried a slightly different method that looked a bit cleaner - let me know what you think, I added a subtype for `StringEncodingError`s that don't take any arguments, makes the show() method for thos just a simple one-liner.